namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala

```diff
@@ -106,14 +108,42 @@ object HttpControlService {
       }
   }
 
+  trait NsPathUri {
+    val prefix: String
+
+    def unapply(uri: String): Option[(Ns, Path)] = {
+      if (uri.startsWith(prefix)) {
+        uri.stripPrefix(prefix).split("/").toSeq match {
+          case ns +: path if path.nonEmpty => Some((ns, Path.Utf8(path: _*)))
+          case _ => None
+        }
+      } else {
+        None
+      }
+    }
+  }
+
+  object BindUri extends NsPathUri {
+    val prefix = "/api/1/bind/"
+  }
+
+  object AddrUri extends NsPathUri {
+    val prefix = "/api/1/addr/"
+  }
+
+  object DelegateUri extends NsPathUri {
+    val prefix = "/api/1/delegate/"
+  }
+
```

Comment:
I suspect that it's a fragile assumption that namespaces can't have slashes...  It would probably be more robust to have the paths be read from a  query param?